### PR TITLE
Added ability to filter enabled or disabled with bin/gpm index

### PR DESCRIPTION
**Pull request by jgonyea**
Third time's the charm (hopefully)

Adds ability to filter index results based on enabled/ disabled settings.

### CODE REVIEW

File: system/src/Grav/Common/GPM/GPM.php [hunk 1]
```diff
@@ -139,6 +139,32 @@ public function getInstalledPlugins()
...
+     */
+    public function isPluginEnabled($slug)
+    {
+        $result = null;
+        $grav = new Grav();
+        $config = $grav::instance()['config']['plugins'];
+        $installed = $this->isPluginInstalled($slug);
+
+        $enabled = isset($config[$slug]) && ((bool)$config[$slug]['enabled'] == true);
+        $disabled = isset($config[$slug]['enabled']) && ((bool)$config[$slug]['enabled'] === false);
```
Comment (mahagr) [inline system/src/Grav/Common/GPM/GPM.php:157]: Umm, aren't these two lines the same?

Here's better code:
```
return (bool)($config[$slug]['enabled'] ?? false);
```

File: system/src/Grav/Common/GPM/GPM.php [hunk 2]
```diff
@@ -139,6 +139,32 @@ public function getInstalledPlugins()
...
+    /**
+     * Returns the plugin's enabled state
+     *
+     * @param  string $slug
+     * @return bool True if the Plugin is Enabled. False if manually set to enable:false. Null otherwise.
+     */
+    public function isPluginEnabled($slug)
+    {
+        $result = null;
+        $grav = new Grav();
```
Comment (mahagr) [inline system/src/Grav/Common/GPM/GPM.php:152]: This won't work. Use `$grav = Grav::instance();`

File: system/src/Grav/Common/GPM/GPM.php [hunk 3]
```diff
@@ -182,6 +208,28 @@ public function getInstalledThemes()
...
+    /**
+     * Checks if a Theme is enabled
+     *
+     * @param  string $slug The slug of the Theme
+     * @return bool
+     *   True if the Theme has been set to the default theme. False if installed, but not enabled. Null otherwise.
+     */
+    public function isThemeEnabled($slug)
+    {
+        $grav = new Grav();
```
Comment (mahagr) [inline system/src/Grav/Common/GPM/GPM.php:220]: Same as above, use instance.

File: system/src/Grav/Common/GPM/GPM.php [hunk 4]
```diff
@@ -182,6 +208,28 @@ public function getInstalledThemes()
...
+        $result = null;
+        $current_theme = $grav::instance()['config']['system']['pages']['theme'];
+        $theme_installed = $this->isThemeInstalled($slug);
+
+        if ($current_theme == $slug) {
```
Comment (mahagr) [inline system/src/Grav/Common/GPM/GPM.php:225]: I'd do the same as with plugins. I would also move check for installed outside of this (and plugin) method and check it elsewhere..

File: system/src/Grav/Console/Gpm/IndexCommand.php [hunk 5]
```diff
@@ -138,7 +150,8 @@ protected function serve(): int
                         'Name' => '<cyan>' . Utils::truncate($package->name, 20, false, ' ', '...') . '</cyan> ',
                         'Slug' => $slug,
                         'Version'=> $this->version($package),
-                        'Installed' => $this->installed($package)
+                        'Installed' => $this->installed($package),
+                        'Enabled' => $this->enabled($package),
```
Comment (mahagr) [inline system/src/Grav/Console/Gpm/IndexCommand.php:154]: and have something like: `'Enabled' => $this->installed($package) ? $this->enabled($package) : '',` here.

**Review by mahagr — CHANGES_REQUESTED** [review]
A few changes are required, others are nice to have.

File: system/src/Grav/Common/GPM/GPM.php (hunk 1 above)
Comment (jgonyea) [reply system/src/Grav/Common/GPM/GPM.php:157]: I think there are technically 3 states:
1. config exists and is set to enabled
2. config exists and is set to disabled
3. config doesn't exist (grav treats this as disabled)

I was attempting to take into consideration the tricky difference between #2 and #3 cases.

File: system/src/Grav/Common/GPM/GPM.php (hunk 2 above)
Comment (jgonyea) [reply system/src/Grav/Common/GPM/GPM.php:152]: I'll fix this.

File: system/src/Grav/Common/GPM/GPM.php (hunk 3 above)
Comment (jgonyea) [reply system/src/Grav/Common/GPM/GPM.php:220]: Same as above...I'll fix this.

File: system/src/Grav/Console/Gpm/IndexCommand.php (hunk 5 above)
Comment (mahagr) [reply system/src/Grav/Console/Gpm/IndexCommand.php:154]: @jgonyea If you check here, this change would check all your 3 cases.

**Pull request merged**